runtime-modules/membership/src/lib.rs

```diff
@@ -1,9 +1,647 @@
+            about: Option<Vec<u8>>
+        ) {
+            // ensure sender is screening authority
+            let sender = ensure_signed(origin)?;
+
+            if <ScreeningAuthority<T>>::exists() {
+                ensure!(sender == Self::screening_authority(), "not screener");
+            } else {
+                // no screening authority defined. Cannot accept this request
+                return Err("no screening authority defined");
+            }
+
+            // make sure we are accepting new memberships
+            ensure!(Self::new_memberships_allowed(), "new members not allowed");
+
+            let user_info = Self::check_user_registration_info(handle, avatar_uri, about)?;
+
+            // ensure handle is not already registered
+            Self::ensure_unique_handle(&user_info.handle)?;
+
+            let member_id = Self::insert_member(&new_member_account, &user_info, EntryMethod::Screening(sender));
+
+            Self::deposit_event(RawEvent::MemberRegistered(member_id, new_member_account));
+        }
+
+        pub fn set_screening_authority(origin, authority: T::AccountId) {
+            ensure_root(origin)?;
+            <ScreeningAuthority<T>>::put(authority);
+        }
+    }
+}
+
+/// Reason why a given member id does not have a given account as the controller account.
+pub enum ControllerAccountForMemberCheckFailed {
+    NotMember,
+    NotControllerAccount,
+}
+
+pub enum MemberControllerAccountDidNotSign {
+    UnsignedOrigin,
+    MemberIdInvalid,
+    SignerControllerAccountMismatch,
+}
+
+pub enum MemberControllerAccountMismatch {
+    MemberIdInvalid,
+    SignerControllerAccountMismatch,
+}
+pub enum MemberRootAccountMismatch {
+    MemberIdInvalid,
+    SignerRootAccountMismatch,
+}
+
+impl<T: Trait> Module<T> {
+    /// Provided that the member_id exists return its membership. Returns error otherwise.
+    pub fn ensure_membership(id: T::MemberId) -> Result<MembershipOf<T>, &'static str> {
+        if <MembershipById<T>>::exists(&id) {
+            Ok(Self::membership(&id))
+        } else {
+            Err("member profile not found")
+        }
+    }
+
+    /// Ensure that given member has given account as the controller account
+    pub fn ensure_is_controller_account_for_member(
+        member_id: &T::MemberId,
+        account: &T::AccountId,
+    ) -> Result<MembershipOf<T>, ControllerAccountForMemberCheckFailed> {
+        if MembershipById::<T>::exists(member_id) {
+            let membership = MembershipById::<T>::get(member_id);
+
+            if membership.controller_account == *account {
+                Ok(membership)
+            } else {
+                Err(ControllerAccountForMemberCheckFailed::NotControllerAccount)
+            }
+        } else {
+            Err(ControllerAccountForMemberCheckFailed::NotMember)
+        }
+    }
+
+    /// Returns true if account is either a member's root or controller account
+    pub fn is_member_account(who: &T::AccountId) -> bool {
```

Comment:
This must go.

Reply:
This is 'allowed check' from the https://github.com/Joystream/joystream/issues/93#issue-522940475 (task item 2).

Reply:
Seeing that point now, sorry for the confusion this caused, this was probably meant to say something like `ensure_*(origin, MemberId)` is ok, but clearly its not. But we do not want this. There should be no constraint on the account space, as this requires, so `is_member_account` should go, and everything that relies on it in the runtime must be updated.